mediatek: add support for Acer Predator W6d

fmorillo opened the pull request:
This commit adds support for Acer Predator W6d.
The device is quite similar to the Acer Predator W6, except is uses two MT7975 for WiFi. And like the W6 a UART connection can be established without opening the device.

This device has secure boot enabled and the bootargs are changed while booting. Additionally a dm-mod.create argument is required. That's why bootargs-override is set.

Hardware
--------
SOC:   MediaTek MT7986A
RAM:   1GB DDR4
FLASH: 4GB eMMC
WiFi:  4x4 802.11 b/g/n/ax MT7975N
       4x4 802.11 a/n/ac/ax MT7975P
ETH:   4x LAN 1Gbit/s (MT7531)
       1x WAN 2.5Gbit/s (GPY211)
BTN:   Reset, WPS
LED:   Antenna LEDs (GPIO)
       Mood-LED (Kinetic KTD2601) - unsupported
UART:  Header Reset to USB port - 3V3 115200 8N1
       [BUTTON] GND - RX - TX [USB]
Power: DC 12V 2A

Installation
------------

1.  Connect to the device using serial console.

2.  Interrupt the autoboot process when prompted by sending '0' twice. In later firmware versions the bootmenu is silenced, in that case the device needs to be powered up with the Reset and WPS button pressed.

3.  Disable the secure boot signature check. $ fdt addr $(fdtcontroladdr) fdt rm /signature

4a. Serve the OpenWrt initramfs image using TFTP at 192.168.1.2. Name
    the image "predator.bin" and connect the TFTP server to the routers
    LAN port.
    $ setenv serverip 192.168.1.2
      setenv ipaddr 192.168.1.1
      tftpboot 0x46000000 predator.bin
      bootm

4b. Or transfer the image over serial with the YMODEM protocol.
    $ picocom -b 115200 --send-cmd 'lrzsz-sz -b -vv' /dev/ttyUSB0
    $ loady 0x46000000 115200
      <CTRL-A><CTRL-S>
      Enter filename
      bootm

5.  Wait for OpenWrt to boot.

6.  Configure U-Boot to allow loading unsigned images from MMC. $ fw_setenv bootcmd 'mmc read 0x46000000 0x00004400 0x0010000; fdt addr $(fdtcontroladdr); fdt rm /signature; bootm'

7.  Configure autoboot by getting all bootmenu entries and set the default entry to max entry + 1. $ fw_printenv fw_setenv bootmenu_default 9

8.  Transfer the OpenWrt sysupgrade image to the router using scp.

9.  Install OpenWrt using sysupgrade.

musashino205 commented on target/linux/mediatek/dts/mt7986a-acer-predator-w6d.dts:
```diff
@@ -0,0 +1,490 @@
+
+
+	leds {
+		compatible = "gpio-leds";
```
`gpio-leds` doesn't use `reg` property, please use `-` instead of `@` for node names of all LEDs.

musashino205 commented on package/boot/uboot-envtools/files/mediatek_filogic:
```diff
@@ -86,6 +86,7 @@ zbtlink,zbt-z8103ax)
 dlink,aquila-pro-ai-m30-a1)
 	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x40000" "0x40000"
 	;;
+acer,predator-w6d|\
```
Please sort alphabetically. (this block should be moved before `asus,rt-ax59u`)

musashino205 commented on target/linux/mediatek/dts/mt7986a-acer-predator-w6d.dts:
Can't a shared dtsi for W6/W6d be created?

fmorillo replied:
I would love to do that, but a few things are different and I'm not able to test that, because I don't have that device.

hauke replied:
You can put the parts which are in both dts files and did not change into a common dtsi file. 
You can check that the `target/linux/mediatek/dts/mt7986a-acer-predator-w6.dts` did not change by decompiling it using the dtc tool.

fmorillo committed:
mediatek: add support for Acer Predator W6d

This commit adds support for Acer Predator W6d.
The device is quite similar to the Acer Predator W6, except is uses
two MT7975 for WiFi. And like the W6 a UART connection can be
established without opening the device.

This device has secure boot enabled and the bootargs are changed while
booting. Additionally a dm-mod.create argument is required. That's why
bootargs-override is set.

Hardware
--------
SOC:   MediaTek MT7986A
RAM:   1GB DDR4
FLASH: 4GB eMMC
WiFi:  4x4 802.11 b/g/n/ax MT7975N
       4x4 802.11 a/n/ac/ax MT7975P
ETH:   4x LAN 1Gbit/s (MT7531)
       1x WAN 2.5Gbit/s (GPY211)
BTN:   Reset, WPS
LED:   Antenna LEDs (GPIO)
       Mood-LED (Kinetic KTD2601) - unsupported
UART:  Header Reset to USB port - 3V3 115200 8N1
       [BUTTON] GND - RX - TX [USB]
Power: DC 12V 2A

Installation
------------

1.  Connect to the device using serial console.

2.  Interrupt the autoboot process when prompted by sending '0' twice.
    In later firmware versions the bootmenu is silenced, in that case the
    device needs to be powered up with the Reset and WPS button pressed.

3.  Disable the secure boot signature check.
    $ fdt addr $(fdtcontroladdr)
      fdt rm /signature

4a. Serve the OpenWrt initramfs image using TFTP at 192.168.1.2. Name
    the image "predator.bin" and connect the TFTP server to the routers
    LAN port.
    $ setenv serverip 192.168.1.2
      setenv ipaddr 192.168.1.1
      tftpboot 0x46000000 predator.bin
      bootm

4b. Or transfer the image over serial with the YMODEM protocol.
    $ picocom -b 115200 --send-cmd 'lrzsz-sz -b -vv' /dev/ttyUSB0
    $ loady 0x46000000 115200
      <CTRL-A><CTRL-S>
      Enter filename
      bootm

5.  Wait for OpenWrt to boot.

6.  Configure U-Boot to allow loading unsigned images from MMC.
    $ fw_setenv bootcmd 'mmc read 0x46000000 0x00004400 0x0010000;
      fdt addr $(fdtcontroladdr); fdt rm /signature; bootm'

7.  Configure autoboot by getting all bootmenu entries and set the
    default entry to max entry + 1.
    $ fw_printenv
      fw_setenv bootmenu_default 9

8.  Transfer the OpenWrt sysupgrade image to the router using scp.

9.  Install OpenWrt using sysupgrade.

Signed-off-by: Fernando Morillo <fernando.m@gmx.de>

goldwrt reviewed (changes requested):
I think, the boot error is a show stopper. The PR should not be committed without an extensively tested modification (serial0 boot log). Even better would be one acer-w6-common.dtsi with specific additions/dts for W6 (MT7916 + GPY21x + KTD2061), W6m (MT7916 + KTD2026), and W6d (MT7975 + GPY21x+ KTD2061).

hauke commented on target/linux/mediatek/image/filogic.mk:
```diff
@@ -161,6 +161,21 @@ define Device/acer_predator-w6
+  DEVICE_DTS := mt7986a-acer-predator-w6d
+  DEVICE_DTS_DIR := ../dts
+  DEVICE_DTS_LOADADDR := 0x47000000
+  DEVICE_PACKAGES := kmod-usb3 kmod-mt7915e kmod-mt7986-firmware mt7986-wo-firmware e2fsprogs f2fsck mkf2fs
```
yes that kernel module is needed for this wifi chip.

jarledo commented on target/linux/mediatek/dts/mt7986a-acer-predator-w6d.dts:
```diff
@@ -0,0 +1,490 @@
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs = "dm-mod.create=\"dm-zero,,,ro,0 1 zero\"";
+		bootargs-override = "root=PARTLABEL=rootfs rootwait";
```
Just to chime in, as someone with no prior experience with openwrt, I just installed this branch rebased on main on two W6d devices. One of them had a 2022.X uboot version and the other one had 2023.X. Both of them seem to be running well for a few days now. For the 2022.X I used `setenv bootmenu_default 7`, while for the 2023.X I used `setenv bootmenu_default 9`. Unfortunately, I forgot to save the serial logs and I've already de-soldered the cables I used for the serial connection. If it's of any help I could solder them back on and post the serial log here when I have the time?

hauke commented:
Please answer the comments and also rebase the PR on current main branch.

hauke commented:
I merged support for this device into main branch using #16861.
